modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/NavigationBar.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayLayout from '@clayui/layout';
+import ClayLink from '@clayui/link';
+import ClayNavigationBar from '@clayui/navigation-bar';
+import classNames from 'classnames';
+import React from 'react';
+
+export default function NavigationBar({cssClass, inverted, navigationItems}) {
+	return (
+		<ClayLayout.ContainerFluid className={classNames('p-0', cssClass)}>
+			<ClayNavigationBar
+				inverted={inverted}
+				triggerLabel={navigationItems.find(({active}) => active)?.label}
+			>
+				{navigationItems.map(({active, href, label}) => (
+					<ClayNavigationBar.Item active={active} key={label}>
+						<ClayLink
+							className="nav-link"
+							displayType="unstyled"
+							{...{href}}
```

Comment:
I'm not clever enough to understand why we are destructuring here :/

Reply:
I'm going to go back to this one final time... but if we generated an `<a href>foo</a>` tag, it would navigate (which many apps don't `preventDefault()`) producing weird effects.

So, the idea is to only pass `href` when there's a value and not when null is passed. Now... the original problem was that we were passing `href=""`, so I'll double check to see if `href={href}` when `href` is null generates or not the attribute.

Reply:
@bruno: It's not destructuring.

- `<Component {...x} />` means "spread the properties from object `x` into the component".
- So, if `x` is `{href: 'hey'}`, then it's equivalent to `<Component href="hey" />`.
- In @jbalsas's code, we don't have an object `x`, we have a variable `href` that may contain a string or `undefined`.
- So, `{href}` there is object shorthand, not destructuring, and it is the same as writing `{href: href}` (ie. equivalent to `{href: undefined}` or `{href: 'some string'}`, depending on the value of `href`).
- In turn, that means that `<ClayLink {...{href}} />` is equivalent to `<ClayLink href={undefined} />` or `<ClayLink href="some string" />`.

And when React renders the `<ClayLink />` and ultimately an `<a>`, if `href` is `undefined` it won't include the attribute in the DOM, and if it is a string, it will include it.

Does that make sense?

Reply:
What @wincent said 👆

Reply:
> So, the idea is to only pass `href` when there's a value and not when null is passed. Now... the original problem was that we were passing `href=""`, so I'll double check to see if `href={href}` when `href` is null generates or not the attribute.

I already checked `undefined` for you in the original PR, but it's true for `null` too:

![Screenshot 2020-06-16 -145838-1NrHVzxf@2x](https://user-images.githubusercontent.com/7074/84777494-f4ee9400-afe1-11ea-933d-81fcdf3962b0.png)

Reply:
You're right, sorry!

So... do we prefer `href={href}` now that that's done?

Reply:
In case you're wondering what the red squiggly in that screenshot is about, it's:

> The href attribute is required for an anchor to be keyboard accessible. Provide a valid, navigable address as the href value. If you cannot provide an href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md (jsx-a11y/anchor-is-valid)eslint

Reply:
That's a good catch! I checked before and `href` is not a mandatory attribute for an anchor, but it does indeed make it unfocusable. 

I'm leaning towards `tabindex="0"` in case `href` is null...?

Reply:
> So... do we prefer `href={href}` or still `{...{href}}` ?

For the answer to that one, you must ask, The Oracle...

![the-oracle](https://user-images.githubusercontent.com/7074/84778410-3df31800-afe3-11ea-98b4-3f73a7a7fc2a.gif)

Now that you changed `href` to never be `''`, that's the important part taken care of. We can use either, and I might have steered you in the wrong direction the other day by doing `{href}` in an example. 🤷‍♂️ 

- Length: tie.
- Readability: `href={href}` wins because we've already seen how the alternative can cause confusion.
- Lint-friendliness: `{...{href}}` wins because it won't run afoul of the linter (arguably, it is sneakily just hiding the problem from the linter; the root issue that the linter is complaining about is still there).

Mostly because of the lint thing, I'd leave it like you have it. Otherwise, I'd go with the more readable one.

Reply:
> @bruno: It's not destructuring.
Oh yeah I mean _spread_ :D What I was wondering was why not just do `href={href}`

Reply:
It's one of those things where in [the original PR](https://github.com/wincent/liferay-portal/pull/356) it looked like this:

```js
const linkAttributes = href ? {href} : {};

<ClayLink {...linkAttributes}>
```

And I thought, can we get rid of that temporary variable and the ternary? So I said, why not this?

```js
<ClayLink {...href}>
```

ie. my mind jumped only as far as getting rid of the variable and the ternary, but it didn't take that extra step to get rid of the entire spread.

Reply:
Not sure this is a bad practice, but I usually do `href="javascript:;"` when I want to do that kind of thing. A lot of people use `href="#"` but that messes with the scroll position.

Reply:
I guess this answers it:

> Anchor elements are often abused as fake buttons by setting their href to # or javascript:void(0) to prevent the page from refreshing, then listening for their click events .
> 
> These bogus href values cause unexpected behavior when copying/dragging links, opening links in a new tab/window, bookmarking, or when JavaScript is loading, errors, or is disabled. They also convey incorrect semantics to assistive technologies, like screen readers.
> 
> Use a <button> instead. In general, you should only use a hyperlink for navigation to a real URL.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a

Reply:
Yup, plus [Deprecating javascript: URLs](https://reactjs.org/blog/2019/08/08/react-v16.9.0.html#deprecating-javascript-urls)

In absence of `href` we should be printing buttons, but i don't want to go through the trouble right now...

Reply:
Yeah, you get a bunch of console spew from those. We got rid of some in https://github.com/liferay/clay/issues/2297 but I still see them on DXP from time to time, so I am sure that some are still lurking around.

Reply:
> [...] but i don't want to go through the trouble right now...

Yeah, right... turns out I've pushed 2 additional commits to render `<button>` instead of `<a>` when `href` is falsy. That way we still have keyboard support by default and keep semantics.

I've also updated `ddm-form-web` where we were querying those elements (and giving an incorrect id).